htdocs/don/class/donstats.class.php

```diff
@@ -83,11 +83,11 @@ public function __construct($db, $socid, $mode, $userid = 0, $typentid = 0, $sta
 		$this->cachefilesuffix = $mode;
 		$this->join = '';
 
-		if ($status === '0' || $status === '1' || $status === '2') {
+		if ($status == '0' || $status == '1' || $status == '2') {
```

Comment:
As $status is an int, comparing to ints is faster - otherwise PHP needs to convert the constant string to an int 😉

Reply:
Except that it currently send string

Reply:
It is a string in practice because you have:

https://github.com/Dolibarr/dolibarr/blob/7ca1a9b5f0ea57b6b8e3d9ebb88b19331519e674/htdocs/don/stats/index.php#L50

I do not know if you need 'intcomma', but if 'int' is enough you can use GETPOSTINT which returns an int.  If not, you should cast the result of GETPOST to int possibly after checks (GETPOSTISSET, etc), but the expected type is int once you feed it further down function calls, etc.